packages/integration-tests/executions.js

```diff
@@ -0,0 +1,143 @@
+  pRetry(
+    async () => {
+      let execution;
+
+      try {
+        const { body } = await executionsApi.getExecutions({ prefix });
+        const executions = JSON.parse(body);
+        execution = executions.results.find(matcher);
+      } catch (err) {
+        throw new pRetry.AbortError(err);
+      }
+
+      if (isNil(execution)) throw new Error('Not Found');
+
+      return execution.arn;
+    },
+    {
+      retries: options.timeout,
+      maxTimeout: 1000
+    }
+  );
+
+/**
+ * Wait for an execution to be completed and return it
+ *
+ * @param {Object} params
+ * @param {string} params.prefix - the prefix configured for the stack
+ * @param {string} params.arn - an execution ARN
+ * @param {Function} params.callback - an async function to invoke the api
+ *   lambda that takes a prefix / user payload. Defaults to
+ *   cumulusApiClient.invokeApifunction to invoke the api lambda
+ * @param {integer} params.timeout - the number of seconds to wait for the
+ *   execution to reach a terminal state. Defaults to 30
+ * @returns {Promise<Object>} the completed execution
+ */
+const getCompletedExecution = async (params = {}) =>
+  pRetry(
+    async () => {
+      let execution;
+
+      try {
+        execution = await executionsApi.getExecution(
+          pick(params, ['prefix', 'arn', 'callback'])
+        );
+      } catch (err) {
+        throw new pRetry.AbortError(err);
+      }
+
+      if (execution.status === 'completed') return execution;
+
+      if (execution.status === 'failed') {
+        throw new pRetry.AbortError(
+          new Error(`Execution ${params.arn} failed`)
+        );
+      }
+
+      if (execution.statusCode === 404) {
+        throw new pRetry.AbortError(
+          new Error(`Execution ${params.arn} not found`)
+        );
+      }
+
+      throw new Error(`Execution ${params.arn} still running`);
+    },
+    {
+      retries: get(params, 'timeout', 30),
+      maxTimeout: 1000
+    }
+  );
+
+/**
+ * Wait for an execution to be failed and return it
+ *
+ * @param {Object} params
+ * @param {string} params.prefix - the prefix configured for the stack
+ * @param {string} params.arn - an execution ARN
+ * @param {Function} params.callback - an async function to invoke the api
+ *   lambda that takes a prefix / user payload. Defaults to
+ *   cumulusApiClient.invokeApifunction to invoke the api lambda
+ * @param {integer} params.timeout - the number of seconds to wait for the
+ *   execution to reach a terminal state. Defaults to 30.
+ * @returns {Promise<Object>} the failed execution
+ */
+const getFailedExecution = async (params = {}) =>
+  pRetry(
+    async () => {
+      let execution;
+
+      try {
+        execution = await executionsApi.getExecution(
+          pick(params, ['prefix', 'arn', 'callback'])
+        );
+      } catch (err) {
+        throw new pRetry.AbortError(err);
+      }
+
+      if (execution.status === 'failed') return execution;
```

Comment:
I think this function and the one above should be merged and renamed to `getExecutionByStatus` or something. And add some logic like:

```
      if (execution.status === expectedStatus) {
        throw new pRetry.AbortError(
          new Error(`Execution ${params.arn} was ${execution.status}, not ${expectedStatus}`)
        );
      }
```